src/main/scala/solr/PersonIndexer.scala

```diff
@@ -101,7 +112,64 @@ object PersonIndexer extends SimpleConversion
     return None
   }
 
-  def buildPerson(uri: String,vivo: Vivo): Option[Person] = {
+   def updateDoc(uri: String,vivo: Vivo): Option[SolrInputDocument] = {
+    buildPerson(uri,vivo,false).foreach{ p =>
+
+      var person:Person = p.copy()
+      val existing = checkExisting(p.uri)
+      
+      if (existing.isDefined && existing.get.updatedAt.isDefined) {
+        // NOTE: need to compare with a person with the same updatedAt value so 
+        // it doesn't diff merely on that field alone
+        val changes:Boolean = hasChanges(existing.get, p.copy(updatedAt=existing.get.updatedAt))
+
+        if (!changes) {
+          // if we are skipping (no changes) reset updated at
+          person = p.copy(updatedAt = existing.get.updatedAt)
+          log.info(String.format("Skipping index for %s. No changes detected", uri))
+       } 
+      }
+
+      val solrDoc = new SolrInputDocument()
+      
+      solrDoc.addField("id",person.uri)
+      solrDoc.addField("alternateId", person.personAttributes.get("alternateId").get)
+      solrDoc.addField("group","people")
+
+      val personJson = person.toJson
+
+      solrDoc.addField("json",personJson)
+      
+      val dateFormatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+      //solrDoc.set("updatedAt",dateFormatter.format(Calendar.getInstance().getTime()));
+      //solrDoc.addField("active_b",false)
+
+      /*HashMap<String, Object> date_value = new HashMap<String, Object>(); 
+      date_value.put("set",dateFormatter.format(Calendar.getInstance().getTime()));
+      solrDoc.addField("updatedAt",date_value)
+
+      HashMap<String, Object> bool_value = new HashMap<String, Object>(); 
+      bool_value.put("set",false); 
+      solrDoc.set("active_b",bool_value)*/
```

Comment:
hi @ht29 it might be good to go back through and delete some of these comments. I could see it being confusing if someone has to come in and troubleshoot.